src/packages/end-to-end/src/__tests__/api/postgres/issues/exogw384.test.ts

```diff
@@ -0,0 +1,99 @@
+import assert from 'assert';
+import { Field, ID, Entity, BaseDataProvider } from '@exogee/graphweaver';
+import Graphweaver from '@exogee/graphweaver-server';
+import gql from 'graphql-tag';
+
+// ESLint, I know it looks like the entities in this file aren't used, but they actually are.
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+@Entity('EntityWithClientGeneratedId', {
+	provider: new BaseDataProvider('EntityWithClientGeneratedId'),
+	apiOptions: { clientGeneratedPrimaryKeys: true },
+})
+class EntityWithClientGeneratedId {
+	@Field(() => ID)
+	id!: string;
+
+	@Field(() => String)
+	description!: string;
+}
+
+const providerWithCreateOne = new BaseDataProvider('EntityWithoutClientGeneratedId');
+providerWithCreateOne.createOne = async () => undefined;
+
+@Entity('EntityWithoutClientGeneratedId', {
+	provider: providerWithCreateOne,
+	apiOptions: { clientGeneratedPrimaryKeys: false },
+})
+class EntityWithoutClientGeneratedId {
+	@Field(() => ID)
+	id!: string;
+
+	@Field(() => String)
+	description!: string;
+}
+
+@Entity('EntityWitClientGeneratedIdNotDefined', {
+	provider: providerWithCreateOne,
+})
+class EntityWitClientGeneratedIdNotDefined {
+	@Field(() => ID)
+	id!: string;
+
+	@Field(() => String)
+	description!: string;
+}
+
+const graphweaver = new Graphweaver();
+
+describe('clientGeneratedPrimaryKeys', () => {
+	test('should throw error because the ID is marked as client-generated, but we are not passing an ID when trying to create a record', async () => {
+		const response = await graphweaver.executeOperation({
+			query: gql`
+				mutation {
+					createEntityWithClientGeneratedId(input: { description: "Test User" }) {
+						id
+						description
+					}
+				}
+			`,
+		});
+
+		assert(response.body.kind === 'single');
+		expect(response.body.singleResult.errors?.[0]?.message).toBe(
+			'Field "EntityWithClientGeneratedIdInsertInput.id" of required type "ID!" was not provided.'
+		);
+	});
```

Comment:
Can we also add a successful case where we `createEntityWithClientGeneratedId` and pass the ID in?

Reply:
@thekevinbrown 
```
mutation {
					createEntityWithClientGeneratedId(input: { id: "12345", description: "Test User" }) {
						id
						description
					}
				}
```

is actually calling `dataProvider.updateOne`, shouldn't it be calling `dataProvider.createOne`?
I know the end-to-end tests are passing and records are being created, but we do have an issue in which we can update a record using the CREATE mutation and passing an existing ID.
Should this be working like that?

Reply:
Yeah, I'd say if you ask for a create and the entity already exists that should error.